src/backend/opencl/CMakeLists.txt

```diff
@@ -1,393 +1,471 @@
-                                PRIVATE ${CBLAS_LIBRARIES})
-ENDIF(LAPACK_FOUND)
 
-LIST(LENGTH GRAPHICS_DEPENDENCIES GRAPHICS_DEPENDENCIES_LEN)
+target_sources(afopencl_static
+  PRIVATE
+    kernel/KParam.hpp
+    kernel/approx.hpp
+    kernel/assign.hpp
+    kernel/bilateral.hpp
+    kernel/canny.hpp
+    kernel/config.cpp
+    kernel/config.hpp
+    kernel/convolve.hpp
+    kernel/convolve_separable.cpp
+    kernel/convolve_separable.hpp
+    kernel/cscmm.hpp
+    kernel/cscmv.hpp
+    kernel/csrmm.hpp
+    kernel/csrmv.hpp
+    kernel/diagonal.hpp
+    kernel/diff.hpp
+    kernel/exampleFunction.hpp
+    kernel/fast.hpp
+    kernel/fftconvolve.hpp
+    kernel/gradient.hpp
+    kernel/harris.hpp
+    kernel/histogram.hpp
+    kernel/homography.hpp
+    kernel/hsv_rgb.hpp
+    kernel/identity.hpp
+    kernel/iir.hpp
+    kernel/index.hpp
+    kernel/interp.hpp
+    kernel/iota.hpp
+    kernel/ireduce.hpp
+    kernel/join.hpp
+    kernel/laset.hpp
+    kernel/laset_band.hpp
+    kernel/laswp.hpp
+    kernel/lookup.hpp
+    kernel/lu_split.hpp
+    kernel/match_template.hpp
+    kernel/meanshift.hpp
+    kernel/medfilt.hpp
+    kernel/memcopy.hpp
+    kernel/moments.hpp
+    kernel/morph.hpp
+    kernel/names.hpp
+    kernel/nearest_neighbour.hpp
+    kernel/orb.hpp
+    kernel/random_engine.hpp
+    kernel/range.hpp
+    kernel/reduce.hpp
+    kernel/regions.hpp
+    kernel/reorder.hpp
+    kernel/resize.hpp
+    kernel/rotate.hpp
+    kernel/scan_dim.hpp
+    kernel/scan_dim_by_key.hpp
+    kernel/scan_dim_by_key_impl.hpp
+    kernel/scan_first.hpp
+    kernel/scan_first_by_key.hpp
+    kernel/scan_first_by_key_impl.hpp
+    kernel/select.hpp
+    kernel/shift.hpp
+    kernel/sift_nonfree.hpp
+    kernel/sobel.hpp
+    kernel/sort.hpp
+    kernel/sort_by_key.hpp
+    kernel/sort_by_key_impl.hpp
+    kernel/sort_helper.hpp
+    kernel/sparse.hpp
+    kernel/sparse_arith.hpp
+    kernel/susan.hpp
+    kernel/swapdblk.hpp
+    kernel/tile.hpp
+    kernel/transform.hpp
+    kernel/transpose.hpp
+    kernel/transpose_inplace.hpp
+    kernel/triangle.hpp
+    kernel/unwrap.hpp
+    kernel/where.hpp
+    kernel/wrap.hpp
+
+    kernel/convolve/conv1.cpp
+    kernel/convolve/conv2_b8.cpp
+    kernel/convolve/conv2_c32.cpp
+    kernel/convolve/conv2_c64.cpp
+    kernel/convolve/conv2_f32.cpp
+    kernel/convolve/conv2_f64.cpp
+    kernel/convolve/conv2_impl.hpp
+    kernel/convolve/conv2_s16.cpp
+    kernel/convolve/conv2_s32.cpp
+    kernel/convolve/conv2_s64.cpp
+    kernel/convolve/conv2_u16.cpp
+    kernel/convolve/conv2_u32.cpp
+    kernel/convolve/conv2_u64.cpp
+    kernel/convolve/conv2_u8.cpp
+    kernel/convolve/conv3.cpp
+    kernel/convolve/conv_common.hpp
+    )
 
-ADD_DEPENDENCIES(afopencl ${cl_kernel_targets})
+target_sources(afopencl_static
+  PRIVATE
+    JIT/BinaryNode.hpp
+    JIT/BufferNode.hpp
+    JIT/Node.hpp
+    JIT/ScalarNode.hpp
+    JIT/UnaryNode.hpp
+  )
+
+target_sources(afopencl_static
+  PRIVATE
+    ${kernel_files}
+  )
```

Comment:
This is not working with clean build because the source files doesn't exist yet at the time of build file generation.